octree/basic_utils_test.go

```diff
@@ -0,0 +1,187 @@
+package octree
+
+import (
+	"context"
+	"testing"
+
+	"github.com/edaniels/golog"
+	"github.com/golang/geo/r3"
+	"github.com/pkg/errors"
+	"go.viam.com/test"
+
+	pc "go.viam.com/rdk/pointcloud"
+)
+
+// Test creation of empty leaf node, filled leaf node and internal node.
+func TestNodeCreation(t *testing.T) {
+	t.Run("Create empty leaf node", func(t *testing.T) {
+		basicOct := newLeafNodeEmpty()
+
+		test.That(t, basicOct.nodeType, test.ShouldResemble, LeafNodeEmpty)
+		test.That(t, basicOct.point, test.ShouldResemble, pc.PointAndData{})
+		test.That(t, basicOct.tree, test.ShouldBeNil)
+	})
+
+	t.Run("Create filled leaf node", func(t *testing.T) {
+		p := r3.Vector{X: 0, Y: 0, Z: 0}
+		d := pc.NewValueData(1.0)
+		basicOct := newLeafNodeFilled(p, d)
+
+		test.That(t, basicOct.nodeType, test.ShouldResemble, LeafNodeFilled)
+		test.That(t, basicOct.point, test.ShouldResemble, pc.PointAndData{P: p, D: d})
+		test.That(t, basicOct.tree, test.ShouldBeNil)
+	})
+
+	t.Run("Create internal node", func(t *testing.T) {
+		var children []*basicOctree
+		basicOct := newInternalNode(children)
+
+		test.That(t, basicOct.nodeType, test.ShouldResemble, InternalNode)
+		test.That(t, basicOct.point, test.ShouldResemble, pc.PointAndData{})
+		test.That(t, basicOct.tree, test.ShouldResemble, children)
+	})
+}
+
+// Tests that the splitting of a filled octree node results in seven empty leaf nodes
+// and one filled one or an empty octree node will result in eight empty children nodes.
+func TestSplitIntoOctants(t *testing.T) {
+	ctx := context.Background()
+	logger := golog.NewTestLogger(t)
+
+	center := r3.Vector{X: 0, Y: 0, Z: 0}
+	side := 1.0
+
+	t.Run("Splitting empty octree node into octants", func(t *testing.T) {
```

Comment:
[q] What is the purpose of being able to split an empty octree node into octants? According to the description of `splitIntoOctants`, trying to split an empty octree node should not be allowed.

Reply:
Originally this was included to allow users to make a basic octree structure if desired, however as this is an unintended use, it has been remove and now will error on empty node